Objective-C/CBLCollection.mm

```diff
@@ -525,7 +525,8 @@ - (BOOL) isEqual: (id)object {
     
     if (!(other && [self.name isEqual: other.name] &&
           [self.scope.name isEqual: other.scope.name] &&
-          [self.database.path isEqual: other.database.path])) {
+          // don't use isEqual: here!  The database must be the exact same instance.
+          self.database == other.database)) {
```

Comment:
Have you seen a problem when comparing the database paths?

Reply:
Yes, with just the first change from `!=` to `![ isEqual:]` it broke [these tests](https://github.com/couchbase/couchbase-lite-ios/actions/runs/7847831080/job/21421330937). The equality check needs to do the check for separate database instances, which then make the collections not equal.